exchanges/kucoin/kucoin_websocket.go

```diff
@@ -0,0 +1,1645 @@
+			return nil, err
+		}
+		if subscriptions[x].Asset == asset.Futures {
+			subscriptions[x].Currency, err = ku.FormatExchangeCurrency(subscriptions[x].Currency, asset.Futures)
+			if err != nil {
+				continue
+			}
+		}
+		switch subscriptions[x].Channel {
+		case marketTickerChannel,
+			marketOrderbookLevel2Channels,
+			marketOrderbookLevel2to5Channel,
+			marketOrderbokLevel2To50Channel,
+			indexPriceIndicatorChannel,
+			marketMatchChannel,
+			markPriceIndicatorChannel:
+			symbols, okay := subscriptions[x].Params["symbols"].(string)
+			if !okay {
+				if subscriptions[x].Currency.IsEmpty() {
+					return nil, errors.New("symbols not passed")
+				}
+				symbols = subscriptions[x].Currency.String()
+			}
+			payloads[x] = WsSubscriptionInput{
+				ID:       strconv.FormatInt(ku.Websocket.Conn.GenerateMessageID(false), 10),
+				Type:     operation,
+				Topic:    fmt.Sprintf(subscriptions[x].Channel, symbols),
+				Response: true,
+			}
+		case marketAllTickersChannel,
+			privateChannel,
+			accountBalanceChannel,
+			marginPositionChannel,
+			spotMarketAdvancedChannel,
+			futuresTradeOrderChannel,
+			futuresStopOrdersLifecycleEventChannel,
+			futuresAccountBalanceEventChannel, futuresSystemAnnouncementChannel:
+			input := WsSubscriptionInput{
+				ID:       strconv.FormatInt(ku.Websocket.Conn.GenerateMessageID(false), 10),
+				Type:     operation,
+				Topic:    subscriptions[x].Channel,
+				Response: true,
+			}
+			switch subscriptions[x].Channel {
+			case futuresTradeOrderChannel,
+				futuresStopOrdersLifecycleEventChannel,
+				futuresAccountBalanceEventChannel:
+				input.PrivateChannel = true
+			}
+			payloads[x] = input
+		case marketTickerSnapshotChannel,
+			futuresPositionChangeEventChannel,
+			futuresTradeOrdersBySymbolChannel,
+			futuresTrasactionStatisticsTimerEventChannel,
+			futuresContractMarketDataChannel,
+			futuresOrderbookLevel2Depth50Channel,
+			futuresOrderbookLevel2Depth5Channel,
+			futuresExecutionDataChannel,
+			futuresOrderbookLevel2Channel,
+			futuresTickerChannel,
+			futuresTickerV2Channel: // Symbols
+			payloads[x] = WsSubscriptionInput{
+				ID:       strconv.FormatInt(ku.Websocket.Conn.GenerateMessageID(false), 10),
+				Type:     operation,
+				Topic:    fmt.Sprintf(subscriptions[x].Channel, subscriptions[x].Currency.String()),
+				Response: true,
+			}
+			switch subscriptions[x].Channel {
+			case futuresPositionChangeEventChannel,
+				futuresTradeOrdersBySymbolChannel:
+				payloads[x].PrivateChannel = true
+			}
+		case marketTickerSnapshotForCurrencyChannel,
+			marginLoanChannel:
+			if subscriptions[x].Channel == marketTickerSnapshotForCurrencyChannel {
+				subscriptions[x].Channel += "%s"
+			}
+			payloads[x] = WsSubscriptionInput{
+				ID:       strconv.FormatInt(ku.Websocket.Conn.GenerateMessageID(false), 10),
+				Type:     operation,
+				Topic:    fmt.Sprintf(subscriptions[x].Channel, subscriptions[x].Currency.Base.Upper().String()),
+				Response: true,
+			}
```

Comment:
Continuing from a previous comment, since you are only sending the currency base, if I have OXEN-BTC,OXEN-ETH enabled, I am sending two duplicate subscription requests

Reply:
🥹 I have added a filter to make these currencies unique